### PR TITLE
add instrumentation for HTTPX

### PR DESCRIPTION
**Pull request by fallwith**
adds instrumentation for [HTTPX](https://honeyryderchuck.gitlab.io/httpx/)

resolves: n/a (Hacktoberfest contribution)

### CODE REVIEW

File: test/multiverse/suites/httpx/httpx_instrumentation_test.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  # HttpClientTestCases required method
+  def timeout_error_class
+    ::NewRelic::LanguageSupport.constantize(NewRelic::Agent::Instrumentation::HTTPX::NOTICEABLE_ERROR_CLASS)
+  end
+
+  # HttpClientTestCases required method
+  def simulate_error_response
+    # TODO
+    # stub something
+    # get_response(default_url)
```
Comment (kaylareopelle) [inline test/multiverse/suites/httpx/httpx_instrumentation_test.rb:100]: Do you want to resolve this TODO before merging?

File: test/multiverse/suites/httpx/httpx_instrumentation_test.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  def initialize(responses = {})
+    @responses = responses
+  end
+end
+
+class HTTPXInstrumentationTest < Minitest::Test
+  include HttpClientTestCases
+
+  # TODO: make sure our transaction level and segment level error
+  #       handling for HTTPX is working as desired
```
Comment (kaylareopelle) [inline test/multiverse/suites/httpx/httpx_instrumentation_test.rb:24]: Do you want to resolve this TODO before merging?

File: lib/new_relic/agent/instrumentation/httpx.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  executes do
+    NewRelic::Agent.logger.info('Installing httpx instrumentation')
+  end
+
+  executes do
+    if use_prepend?
+      prepend_instrument HTTPX::Session, NewRelic::Agent::Instrumentation::HTTPX::Prepend, 'HTTPX'
+    else
+      chain_instrument NewRelic::Agent::Instrumentation::HTTPX::Chain, 'HTTPX'
```
[the remark this answers is not in the document]
Comment (fallwith) [reply lib/new_relic/agent/instrumentation/httpx.rb]: Ah, thank you! Fixed here with [ee243b9](https://github.com/newrelic/newrelic-ruby-agent/pull/2278/commits/ee243b97be5fe34ebd43b977588603cbf2b027e2) and updated the Ethon PR comment accordingly as well.

File: test/multiverse/suites/httpx/httpx_instrumentation_test.rb (hunk 1 above)
Comment (fallwith) [reply test/multiverse/suites/httpx/httpx_instrumentation_test.rb]: Error handling and testing updated with [ee243b9](https://github.com/newrelic/newrelic-ruby-agent/pull/2278/commits/ee243b97be5fe34ebd43b977588603cbf2b027e2), no more TODOs remain. Thank you!

File: test/multiverse/suites/httpx/httpx_instrumentation_test.rb (hunk 2 above)
Comment (fallwith) [reply test/multiverse/suites/httpx/httpx_instrumentation_test.rb]: Error handling and testing updated with [ee243b9](https://github.com/newrelic/newrelic-ruby-agent/pull/2278/commits/ee243b97be5fe34ebd43b977588603cbf2b027e2), no more TODOs remain. Thank you!

**Review by kaylareopelle — DISMISSED** [review]
Thanks for your contribution, @fallwith! 😄 🎃

**Review by kaylareopelle — APPROVED** [review]

**Pull request merged**